### PR TITLE
Feature/functionaltesting

### PR DESCRIPTION
**Pull request by Toukakoukan**
Again quite a large one.
Biggest additional was functional testing of `snapraid sync` within TestKitchen. 
This involved
- Addition of a vagrantfile.rb that adds 3 x virtual disks (this will get merged into each TestKitchen Vagrant file)
- Addition of a `pre_create_command` in `.kitchen.yml` to delete the 3 x virtual disks prior to recreation
- Addition of a `test_helper.bash` which partitions, formats, and mounts the disks
- Addition of a test which runs `sudo /usr/local/bin/snapraid --force-empty sync`

I also moved the installation specific defaults for 
- `data_disks`
- `parity_disks`
- `content_files`  

out of `attributes\default.rb` and put them in `.kitchen.yml` and as an example in the readme.

In order to get the above functional testing for `snapraid sync` to work I 
- Moved `/etc/snapraid/snapraid.conf` to `/etc/snapraid.conf`
- Amended multiple parity names to be prepended with *n*-parity
- Added `snapraid.content` to the filepath of content files to match opinionated `parity_disks` file naming
- Added `snapraid.content` as a default exclusion.

I also fixed the rakefile (which was obscuring RuboCop's exit code and causing Travis CI to ignore failures) and added a `changelog.md`

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 driver:
   name: vagrant
+  vagrantfiles: 
+      - vagrantfile.rb
```
Comment (X0nic) [inline .kitchen.yml]: Just curious. Why the additional vagrantfile? You can add ruby to the regular `Vagrantfile`. Feel free to modify that one if you need to.

File: vagrantfile.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+
+Vagrant.configure('2')  do |config|
```
Comment (X0nic) [inline vagrantfile.rb]: I really like this. Makes the test kitchen tests a lot more accurate. :)

**Comment by X0nic** [comment]
I really like this addition. You really went all out on the testing. :)

I just had a question about the vagrantfile.rb, other than that it looks good to me.

File: .kitchen.yml (hunk 1 above)
Comment (Toukakoukan) [inline .kitchen.yml]: Hi Nathan
I don't think I can point test kitchen at an entire vagrantfile, as it's just added into the ephemeral test vagrantfile as a require and therefore needs to have the .rb extension. 
Do we still need the original vagrantfile, or should we just use test kitchen?

What do you think?

**Comment by Toukakoukan** [comment]
Hi Nathan
I don't think I can point test kitchen at an entire vagrantfile, as it's just added into the ephemeral test vagrantfile as a require and therefore needs to have the .rb extension. 
Do we still need the original vagrantfile, or should we just use test kitchen?

What do you think?

File: .kitchen.yml (hunk 1 above)
Comment (X0nic) [inline .kitchen.yml]: I don't think the other vagrantfile is needed. I would just overwrite it.

Comment (Toukakoukan) [inline .kitchen.yml]: Happy if I kill the other but keep the .rb one named as is? If l capitalise it rubocop will complain, if I drop the .rb test kitchen will complain :)

Comment (X0nic) [inline .kitchen.yml]: Ok, leave the .rb in. The other one isnt needed, we should kill it.

Comment (Toukakoukan) [inline .kitchen.yml]: Done!
Thanks for checking through all this and discussing Nathan :)

**Comment by X0nic** [comment]
Awesome! thanks for putting up with me. :)

**Comment by Toukakoukan** [comment]
Haha, together we'll make this cookbook tip top :)
Only two more PRs to cast your eyes over and then you're rid of me until I get round to adding Windows support!

**Pull request merged**